lib/internal/modules/cjs/loader.js

```diff
@@ -435,108 +420,28 @@ function trySelfParentPath(parent) {
-  if (typeof exports !== 'object')
-    return false;
-  let isConditional = false;
-  let firstCheck = true;
-  for (const key of ObjectKeys(exports)) {
-    const curIsConditional = key[0] !== '.';
-    if (firstCheck) {
-      firstCheck = false;
-      isConditional = curIsConditional;
-    } else if (isConditional !== curIsConditional) {
-      throw new ERR_INVALID_PACKAGE_CONFIG(basePath, '"exports" cannot ' +
-          'contain some keys starting with \'.\' and some not. The exports ' +
-          'object must either be an object of package subpath keys or an ' +
-          'object of main entry condition name keys only.');
-    }
-  }
-  return isConditional;
-}
-
-function applyExports(basePath, expansion) {
-  const mappingKey = `.${expansion}`;
-
-  let pkgExports = readPackageExports(basePath);
-  if (pkgExports === undefined || pkgExports === null)
-    return false;
-
-  if (isConditionalDotExportSugar(pkgExports, basePath))
-    pkgExports = { '.': pkgExports };
-
-  if (typeof pkgExports === 'object') {
-    if (ObjectPrototypeHasOwnProperty(pkgExports, mappingKey)) {
-      const mapping = pkgExports[mappingKey];
-      const resolved = resolveExportsTarget(
-        pathToFileURL(basePath + '/'), mapping, '', mappingKey);
-      if (resolved === null || resolved === undefined)
-        throw new ERR_PACKAGE_PATH_NOT_EXPORTED(
-          basePath, mappingKey);
-      return resolved;
-    }
-
-    let dirMatch = '';
-    for (const candidateKey of ObjectKeys(pkgExports)) {
-      if (candidateKey[candidateKey.length - 1] !== '/') continue;
-      if (candidateKey.length > dirMatch.length &&
-          StringPrototypeStartsWith(mappingKey, candidateKey)) {
-        dirMatch = candidateKey;
-      }
-    }
-
-    if (dirMatch !== '') {
-      const mapping = pkgExports[dirMatch];
-      const subpath = StringPrototypeSlice(mappingKey, dirMatch.length);
-      const resolved = resolveExportsTarget(pathToFileURL(basePath + '/'),
-                                            mapping, subpath, mappingKey);
-      if (resolved === null || resolved === undefined)
-        throw new ERR_PACKAGE_PATH_NOT_EXPORTED(
-          basePath, mappingKey + subpath);
-      // Extension searching for folder exports only
-      const rc = stat(resolved);
-      if (rc === 0) return resolved;
-      if (!(RegExpPrototypeTest(trailingSlashRegex, resolved))) {
-        const exts = ObjectKeys(Module._extensions);
-        const filename = tryExtensions(resolved, exts, false);
-        if (filename) return filename;
-      }
-      if (rc === 1) {
-        const exts = ObjectKeys(Module._extensions);
-        const filename = tryPackage(resolved, exts, false,
-                                    basePath + expansion);
-        if (filename) return filename;
-      }
-      // Undefined means not found
-      return;
-    }
+  try {
+    return finalizeEsmResolution(packageExportsResolve(
+      pathToFileURL(pkgPath + '/package.json'), expansion, pkg,
```

Comment:
using a forward slash here works on windows as well?

Reply:
Yes, `/` in windows paths works fine and is actually a good pattern for many cases unless dropping paths into exec calls :)

Reply:
Yeah, that should work just fine especially since it's a path being converted to a file URL.